cpp/src/parquet/arrow/reader.cc

```diff
@@ -1025,12 +1025,15 @@ Result<std::unique_ptr<RecordBatchReader>> FileReaderImpl::GetRecordBatchReader(
   ::arrow::Iterator<RecordBatchIterator> batches = ::arrow::MakeFunctionIterator(
       [readers, batch_schema, num_rows,
        this]() mutable -> ::arrow::Result<RecordBatchIterator> {
-        ::arrow::ChunkedArrayVector columns(readers.size());
-
         // don't reserve more rows than necessary
         int64_t batch_size = std::min(properties().batch_size(), num_rows);
-        num_rows -= batch_size;
+        if (batch_size == 0) {
+          // We can return end immediately for 0 batch size
+          return ::arrow::IterationTraits<RecordBatchIterator>::End();
+        }
```

Comment:
I'm not familiar with this reader implementation but can we do this optimization?
If we can do this, we can assume that row group/column chunk metadata are available in data load logic.

Reply:
Yes, this fixes the issue from calling `FileReader::GetRecordBatchReader`. However, is it cleaner to fix it in the `LeafReader::LoadBatch`? There might be use cases that directly use the `ColumnReader` interface by calling `FileReader::GetColumn` instead of `FileReader::GetRecordBatchReader`.

Reply:
Agree. `Status LoadBatch(int64_t records_to_read) final` has called `AttachStatistics`, which forces statistics exists, and the metadata will also calles rowgroup to call `input_->column_chunk_metadata()`

Reply:
It makes sense.

I'll change the implementation.

Reply:
Done.